azurelinuxagent/ga/exthandlers.py

```diff
@@ -728,17 +733,30 @@ def report_ext_handlers_status(self):
         """
         vm_status = VMStatus(status="Ready", message="Guest Agent is running",
                              gs_aggregate_status=self.__gs_aggregate_status)
-        if self.ext_handlers is not None:
-            for ext_handler in self.ext_handlers.extHandlers:
+
+        handlers_to_report = []
+
+        # Incase of Unsupported error, report the status of the handlers in the VM
+        if self.__last_gs_unsupported():
+            for item, path in list_agent_lib_directory(skip_agent_package=True):
                 try:
-                    self.report_ext_handler_status(vm_status, ext_handler)
-                except ExtensionError as error:
-                    add_event(
-                        AGENT_NAME,
-                        version=CURRENT_VERSION,
-                        op=WALAEventOperation.ExtensionProcessing,
-                        is_success=False,
-                        message=ustr(error))
+                    handler_instance = ExtHandlersHandler.get_ext_handler_instance_from_path(name=item,
+                                                                                             path=path,
+                                                                                             protocol=self.protocol)
+                    if handler_instance is not None:
+                        handlers_to_report.append(handler_instance.ext_handler)
+                except Exception:
+                    continue
```

Comment:
do we need to log the error here?

Reply:
no not really needed, this would throw in very rare cases where we're unable to parse the path and convert it to an ExtHandler object, not to mention that this code would be called every 6 secs so logging would fill up the logs really fast. I think its fine as is, worst case we won't report the buggy handler.

Reply:
If this can be produced by a buggy handler then I think we should definitively report it (addressing the verbosity issue)

Reply:
Sure, added a change to report this only on incarnation change.